syntax = "proto3";

package com.nus.dbsytem.falcon.v0.inference;

// The logistic regression inference service definition.
service InferenceLR {
  // Sends a greeting
  rpc Prediction (PredictionRequest) returns (PredictionResponse) {}
}

// The request message containing the prediction sample ids
message PredictionRequest {
  // number of samples in the request
  int32 sample_num = 1;
  // sample ids in the request
  repeated int32 sample_ids = 2;
}

// The prediction output of each sample
message PredictionOutput {
  // predicted label
  float label = 1;
  // prediction confidence scores
  repeated float probabilities = 2;
}

// The response message containing the prediction outputs
message PredictionResponse {
  // number of samples in the response
  int32 sample_num = 1;
  // prediction output in the response
  repeated PredictionOutput outputs = 2;
}
